### ISSUE TITLE
tf.nn.conv2d with invalid input dims crashes in TF ≤2.19 — now raises InvalidArgumentError in nightly

### ISSUE THREAD

**Issue by panda123dd** [tensorflow/tensorflow#95415]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.19

### Custom code

Yes

### OS platform and distribution

linux ubuntu 22.04

### Python version

3.9

### Current behavior?

On Ubuntu 22.04 with TensorFlow 2.19 (stable branch), running the following code:

```python
import tensorflow as tf
x = tf.random.normal([4, 10], mean=1.75, stddev=0.5)
initializer = tf.initializers.truncated_normal(mean=0.0, stddev=0.05)
w = tf.Variable(initializer([10, 5]), dtype=tf.float32)
b = tf.Variable(tf.zeros([5]), dtype=tf.float32)
y = tf.nn.softmax(tf.matmul(x, w) + b)
y_conv = tf.nn.conv2d(x, w, strides=[1, 1, 1, 1], padding='VALID')

results in a crash with abort/core dumped, showing:
Check failed: index >= 0 && index < num_total_dims Invalid index from the dimension: 3，0，C
Aborted (core dumped)

However, in TensorFlow 2.20.0-dev20250604 (nightly), the same code raises a catchable InvalidArgumentError:
convolution input must be 4-dimensional: [4,10] [Op:Conv2D] name: 


### Standalone code to reproduce the issue

```shell
import tensorflow as tf
x = tf.random.normal([4, 10], mean=1.75, stddev=0.5)
initializer = tf.initializers.truncated_normal(mean=0.0, stddev=0.05)
w = tf.Variable(initializer([10, 5]), dtype=tf.float32)
b = tf.Variable(tf.zeros([5]), dtype=tf.float32)
y = tf.nn.softmax((tf.matmul(x, w) + b))
y_conv = tf.nn.conv2d(x, w, strides=[1, 1, 1, 1], padding='VALID')
```

### Relevant log output

```shell
Check failed: index >= 0 && index < num_total_dims Invalid index from the dimension: 3，0，C
Aborted (core dumped)
```

**Comment by Venkat6871** [comment]
Hi @panda123dd ,
Hi, thanks for raising your concern!
I tested your code on both TensorFlow 2.19.0 and a nightly version using Google Colab. In both versions, the code does not crash, but instead correctly raises an error due to an input rank mismatch for `tf.nn.conv2d`, which expects a 4D tensor as input.
As expected:
`x` has shape `[4, 10]` (i.e., it is a 2D tensor),
But `tf.nn.conv2d` requires a 4D input of the form `[batch, height, width, channels]`.

To fix the issue, you can reshape x into a 4D tensor. I made this change and the code worked fine for me. Here is a [Gist](https://colab.sandbox.google.com/gist/Venkat6871/554482d6c1223feeab217dfc7ea71dde/95415_tf_2-19-0-nightly-v.ipynb) for your reference.

**Note**: The nightly version of TensorFlow changes frequently, so for more stable and consistent behavior, it is recommended to stick with official stable releases.

Thank you!
